bookserver/crud.py

```diff
@@ -118,10 +127,118 @@ async def fetch_course(course_name: str) -> CoursesValidator:
     query = select(AuthUser).where(AuthUser.username == user_name)
     async with async_session() as session:
         res = await session.execute(query)
         rslogger.debug(f"res = {res}")
     user = res.scalars().first()
     return AuthUserValidator.from_orm(user) if user else None
+
+
+async def create_user(user: AuthUserValidator) -> int:
+    """
+    The given user will have the password in plain text.  First we will hash
+    the password then add this user to the database.
+    """
+    new_user = AuthUser(**user.dict())
+    print(settings.web2py_private_key)
+    crypt = CRYPT(key=settings.web2py_private_key, salt=True)
+    new_user.password = str(crypt(user.password)[0])
+    res = None
+    try:
+        async with async_session() as session:
+            res = session.add(new_user)
+            await session.commit()
+    except IntegrityError:
+        rslogger.error("Failed to add a duplicate user")
+
+    return AuthUserValidator.from_orm(res) if res else None
+
+
+# instructor_courses
+# ------------------
+async def fetch_instructor_courses(
+    instructor_id: int, course_id: Optional[int] = None
+) -> List[CourseInstructorValidator]:
+    """
+    return a list of courses for which the given userid is an instructor.
+    If the optional course_id value is included then retur the row for that
+    course to verify that instructor_id is an instructor for course_id
+    """
+
+    query = select(CourseInstructor)
+    if course_id is not None:
+        query = query.where(
+            and_(
+                CourseInstructor.instructor == instructor_id,
+                CourseInstructor.course == course_id,
+            )
+        )
+    else:
+        query = query.where(CourseInstructor.instructor == instructor_id)
+    async with async_session() as session:
+        res = await session.execute(query)
+
+    course_list = [
+        CourseInstructorValidator.from_orm(x) for x in res.scalars().fetchall()
+    ]
+    return course_list
+
+
+async def create_initial_courses_users():
+    if settings.drop_tables == "Yes":
+        print("Populating Courses")
+        BASE_COURSES = [
+            "ac1",
+            "cppds",
+            "cppforpython",
+            "csawesome",
+            "csjava",
+            "fopp",
+            "httlads",
+            "java4python",
+            "JS4Python",
+            "learnwebgl2",
+            "MasteringDatabases",
+            "overview",
+            "py4e-int",
+            "pythonds",
+            "pythonds3",
+            "StudentCSP",
+            "TeacherCSP",
+            "thinkcpp",
+            "thinkcspy",
+            "webfundamentals",
+        ]
+
+        for c in BASE_COURSES:
+            new_course = CoursesValidator(
+                course_name=c,
+                base_course=c,
+                term_start_date=datetime.date(2000, 1, 1),
+            )
+            await create_course(new_course)
+        # make a user
+        await create_user(
```

Comment:
I'd suggest moving this into the test code.

Reply:
So much development relies on interactive work.  This makes it possible to get a working environment quickly without running a bunch of manual commands to get things populated.

Reply:
It is kind of like the `rsmanage initdb`